blog/project-portal-completed.md

```diff
@@ -0,0 +1,140 @@
+* **Limited scalability**: The single repository would get more and more unwieldy as new jurisdictions were added, and if one of them had totally different requirements (like having to be hosted on a different service) then we would have to fork the repository and maintain duplicate codebases. 
+* **Violates [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)**: The content management system would give the government staff managing each site permissions to modify the common git repository. If the single repository approach was kept, there was a chance that a modification by staff for one site might inadvertently break something for _all_ the sites in that same repository.
+
+Instead, we decided to split the code across several repositories:
+* A monorepo with the "business logic." We would publish packages of this code, and those packages would be installed independently.
+* Independent repositories for each of the sites.
+
+Over the course of 2022 we rebuilt the sites from the ground up, restructuring the code from the prototype and adding new functionality. We published three new packages:
+* The Project Portal "Theme" which coded all the styling and behavior of the website, and defined an interface on the [Gatsby Data Layer](https://www.gatsbyjs.com/docs/reference/graphql-data-layer/),
+* A data plugin to handle loading data from Airtable, and
+* An alternative data plugin to handle saving and loading data using [Netlify CMS](https://www.netlifycms.org/), which saves its data into the git repository from which the site is built.
+
+We had planned to transition the sites to using the new packages step by step: 1) migrate from a single repo to multiple repositories but keep Airtable for data storage, then 2) switch from the Airtable plugin to the Netlify CMS plugin. 
+
+However, after testing our new implementation, we decided it was easier and less risky to migrate the websites in a [“Big Bang”](https://en.wikipedia.org/wiki/Big_bang_adoption), switching from the old single-repo with Airtable setup to the new multi-repo with Netlify CMS setup. The Airtable plugin was never used in production.
+
+The final structure of the site's data flow is shown in the figure below.
+
+<figure>
+  <img 
+    src="/content/images/blog/project-portal-completed/data-flow.png" 
+    alt="Data Flow diagram of the portal, showing how data stored in the GitHub repository are loaded by the CMS plugin into the GatsbyJS Data Layer, from where the theme plugin builds the deployed site. This all happens using GatsbyJS, running on Netlify." 
+    width=1368 
+    height=488
+  />
+  <figcaption>
+    Data flow of the portal from storage on GitHub to the deployed site.
+  </figcaption>
+</figure>
+
+
+During the course of the development, we encountered some challenges, and learned some lessons.
```

Comment:
Do you think this should be separated into a sub section or even its own section for "Our learnings" from the project or something like that. Just a thought